sourcespec/ssp_build_spectra.py

```diff
@@ -169,8 +169,32 @@ def _cut_signal_noise(config, trace):
     # ...taper...
     cosine_taper(trace_signal.data, width=config.taper_halfwidth)
     cosine_taper(trace_noise.data, width=config.taper_halfwidth)
+
+    # Be sure that both traces have same length:
```

Comment:
I find these lines confusing, is the following code equivalent to yours ?

```python
    # Be sure that both traces have same length:
    npts = min(len(trace_signal), len(trace_noise))
    if len(trace_signal) <= len(trace_noise):
        # truncate noise to signal length
        trace_noise.data = trace_noise.data[:npts]
    else:
        if config.weighting == 'noise':
            msg = '{}: Truncating signal window to noise length!'
            msg = msg.format(trace_signal.get_id()[0:-1])
            logger.warning(msg)
            trace_signal.data = trace_signal.data[:npts]
        else:
            # Zero-pad noise to signal length
            # Note: no risk of ringing, as noise has been tapered
            pad_len = len(trace_signal) - len(trace_noise)
            pad_before = np.floor(pad_len / 2.)
            pad_after = np.ceil(pad_len / 2.)
            time_delta = trace_noise.stats.delta
            t1 = trace_noise.stats.starttime - pad_before * time_delta
            t2 = trace_noise.stats.endtime + pad_after * time_delta
            trace_noise.trim(starttime=t1, endtime=t2, pad=True, fill_value=0)
            msg = '{}: Zero-padding noise window to signal length'
            msg = msg.format(trace_noise.get_id()[0:-1])
            logger.warning(msg)
```

Reply:
Yes, I agree this is more clear. Should I adapt it in my branch or will you do the modification?

Reply:
You can do the commit yourself. Here's a slightly improved version:
```python
    # Be sure that both traces have same length:
    npts = min(len(trace_signal), len(trace_noise))
    if len(trace_signal) <= len(trace_noise):
        # truncate noise to signal length
        trace_noise.data = trace_noise.data[:npts]
    else:
        if config.weighting == 'noise':
            msg = '{}: Truncating signal window to noise length!'
            trace_signal.data = trace_signal.data[:npts]
        else:
            msg = '{}: Zero-padding noise window to signal length'
            # Note: no risk of ringing, as noise has been tapered
            pad_len = len(trace_signal) - len(trace_noise)
            pad_before = np.floor(pad_len / 2.)
            pad_after = np.ceil(pad_len / 2.)
            time_delta = trace_noise.stats.delta
            t1 = trace_noise.stats.starttime - pad_before * time_delta
            t2 = trace_noise.stats.endtime + pad_after * time_delta
            trace_noise.trim(starttime=t1, endtime=t2, pad=True, fill_value=0)
        msg = msg.format(trace_noise.get_id()[0:-1])
        logger.warning(msg)
```

Reply:
Just a doubt: are you sure that `trace_noise.trim()` pads to exactly the same number of points of signal length? Since the padding is based on time, there might be round-off errors.

Would'n it be better to use [np.pad()](https://numpy.org/doc/stable/reference/generated/numpy.pad.html)?

Reply:
I didn't encounter any problems yet, but I only tested the waveforms of 1 earthquake so far.
So, with np.pad, this should read:
`trace_noise.data = np.pad(trace_noise.data, (pad_before, pad_after))`
To be entirely correct, we should also update the trace header manually:
`trace_noise.stats.starttime = t1`
`trace_noise.stats.npts = len(trace_signal)`
Is this correct?

Reply:
Only the start time. When you update the data vector, `npts` and `endtime` are automatically updated.

But, why not padding just at the end?

Reply:
If you want to use `pad_before` and `pad_after`, I think that the following is safer against rounding errors:

```python
            pad_before = np.floor(pad_len / 2.)
            pad_after = pad_len - pad_before
```

But, again, why not padding just at the end?

Reply:
Padding on both sides appears cleaner to me, but if it doesn't make a difference for calculating the Fourier amplitude spectrum, then just padding at the end is indeed more simple, as it doesn't require manually changing trace header words (which I just tested).

Reply:
OK, I will do a quick test.

Reply:
I see no difference between symmetric and one-sided padding.
![pad_fas_comparison](https://user-images.githubusercontent.com/2128835/210763283-d0d9a60d-b39b-45bc-b03e-3b7914cfaad2.png)

Reply:
Great! Let's make it simple then!

Reply:
OK, done.